pkg/operator/starter.go

```diff
@@ -127,39 +127,17 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 
 	// Create GenericOperatorClient. This is used by the library-go controllers created down below
 	// Operator CR is in the guest cluster.
-	extractApplySpec := func(obj *unstructured.Unstructured, fieldManager string) (*applyoperatorv1.OperatorSpecApplyConfiguration, error) {
-		castObj := &opv1.ClusterCSIDriver{}
-		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, castObj); err != nil {
-			return nil, fmt.Errorf("unable to convert to ClusterCSIDriver: %w", err)
-		}
-		ret, err := applyoperatorv1.ExtractClusterCSIDriver(castObj, fieldManager)
-		if err != nil {
-			return nil, fmt.Errorf("unable to extract fields for %q: %w", fieldManager, err)
-		}
-		if ret.Spec == nil {
-			return nil, nil
-		}
-		return &ret.Spec.OperatorSpecApplyConfiguration, nil
-	}
-	extractApplyStatus := func(obj *unstructured.Unstructured, fieldManager string) (*applyoperatorv1.OperatorStatusApplyConfiguration, error) {
-		castObj := &opv1.ClusterCSIDriver{}
-		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, castObj); err != nil {
-			return nil, fmt.Errorf("unable to convert to ClusterCSIDriver: %w", err)
-		}
-		ret, err := applyoperatorv1.ExtractClusterCSIDriverStatus(castObj, fieldManager)
-		if err != nil {
-			return nil, fmt.Errorf("unable to extract fields for %q: %w", fieldManager, err)
-		}
-
-		if ret.Status == nil {
-			return nil, nil
-		}
-		return &ret.Status.OperatorStatusApplyConfiguration, nil
-	}
-
 	gvr := operatorv1.SchemeGroupVersion.WithResource("csisnapshotcontrollers")
 	gvk := operatorv1.SchemeGroupVersion.WithKind("CSISnapshotControllers")
-	guestOperatorClient, dynamicInformers, err := goc.NewClusterScopedOperatorClientWithConfigName(clock.RealClock{}, guestKubeConfig, gvr, gvk, "cluster", extractApplySpec, extractApplyStatus)
+	guestOperatorClient, dynamicInformers, err := goc.NewClusterScopedOperatorClientWithConfigName(
+		clock.RealClock{},
+		guestKubeConfig,
+		gvr,
+		gvk,
+		string(opv1.GCPPDCSIDriver),
```

Comment:
this should be "cluster", not "pd.csi.storage.gke.io".

Reply:
thanks a lot!